packages/workspace/src/utils/rules/rename-package-imports.ts

```diff
@@ -0,0 +1,72 @@
+import { SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getWorkspace } from '@nrwl/workspace';
+import { getFullProjectGraphFromHost } from '@nrwl/workspace/src/utils/ast-utils';
+
+export interface PackageNameMapping {
+  [packageName: string]: string;
+}
+
+const getProjectNamesWithDepsToRename = (
+  packageNameMapping: PackageNameMapping,
+  tree: Tree
+) => {
+  const packagesToRename = Object.entries(packageNameMapping);
+  const projectGraph = getFullProjectGraphFromHost(tree);
+
+  return Object.entries(projectGraph.dependencies)
+    .filter(([, deps]) =>
+      deps.some(
+        (dep) =>
+          dep.type === 'static' &&
+          packagesToRename.some(
+            ([packageName]) => packageName === dep.target.replace('npm:', '')
+          )
+      )
+    )
+    .map(([projectName]) => projectName);
+};
+
+/**
+ * Updates all the imports found in the workspace
+ *
+ * @param packageNameMapping The packageNameMapping provided to the schematic
+ */
+export function renamePackageImports(packageNameMapping: PackageNameMapping) {
+  return async (tree: Tree, _context: SchematicContext): Promise<void> => {
+    const workspace = await getWorkspace(tree);
+
+    const projectNamesThatImportAPackageToRename = getProjectNamesWithDepsToRename(
+      packageNameMapping,
+      tree
+    );
+
+    const projectsThatImportPackage = [...workspace.projects].filter(([name]) =>
+      projectNamesThatImportAPackageToRename.includes(name)
+    );
+
+    projectsThatImportPackage
+      .map(([, definition]) => tree.getDir(definition.root))
+      .forEach((projectDir) => {
+        // only look at .(j|t)s(x) files
+        projectDir.visit((file) => {
+          if (!/(j|t)sx?$/.test(file)) {
+            return;
+          }
+          const contents = tree.read(file).toString('utf-8');
+          Object.entries(packageNameMapping).forEach(
+            ([packageName, newPackageName]) => {
+              if (!contents.includes(packageName)) {
+                return;
+              }
+              // TODO: use AST to make change instead of string replacement
+              const updatedFile = tree
```

Comment:
So, I left this TODO in because I didn't have time to take a crack at learning how to do this properly with the AST. Not sure if you'd like to take a stab at it, or wait on it, or let it go with this simple string replacement

Reply:
I have replaced the simple string replacement with a safer AST replacement